perllib/FixMyStreet/Script/UpdateAllReports.pm

```diff
@@ -66,6 +67,21 @@ sub generate {
                 $open{$cobrand}{$body}{$type}++;
             }
         }
+
+        if ( FixMyStreet->config('AREAS_IN_ALLREPORTS_JSON') ) {
+            @areas = grep { $_ } split( /,/, $problem{areas} );
+            foreach my $area ( @areas ) {
+                my $duration_str = ( $problem{duration} > 2 * $fourweeks ) ? 'old' : 'new';
+                my $type = ( $problem{duration} > 2 * $fourweeks )
+                    ? 'unknown'
+                    : ($problem{age} > $fourweeks ? 'older' : 'new');
+                if (FixMyStreet::DB::Result::Problem->fixed_states()->{$problem{state}} || FixMyStreet::DB::Result::Problem->closed_states()->{$problem{state}}) {
```

Comment:
As body loop has this conditional too, worth sticking it in a variable first and using?